lib/parachain/collation_protocol.go

```diff
@@ -96,3 +101,83 @@ type CollationSeconded struct {
 func (CollationSeconded) Index() uint {
 	return 4
 }
+
+const MaxCollationMessageSize uint64 = 100 * 1024
+
+type CollationProtocolV1 struct{}
+
+// Type returns CollationMsgType
+func (*CollationProtocolV1) Type() network.MessageType {
+	return network.CollationMsgType
+}
+
+// Hash returns the hash of the CollationProtocolV1
+func (cp *CollationProtocolV1) Hash() (common.Hash, error) {
+	// scale encode each extrinsic
+	encMsg, err := cp.Encode()
+	if err != nil {
+		return common.Hash{}, fmt.Errorf("cannot encode message: %w", err)
+	}
+
+	return common.Blake2bHash(encMsg)
+}
+
+// Encode a collator protocol message using scale encode
+func (cp *CollationProtocolV1) Encode() ([]byte, error) {
+	enc, err := scale.Marshal(*cp)
+	if err != nil {
+		return nil, err
+	}
+	return enc, nil
+}
+
+func decodeCollationMessage(in []byte) (network.NotificationsMessage, error) {
```

Comment:
is there a reason we need a wrapper around unmarshal here?

Reply:
RegisterNotificationsProtocol accepts a certain type of decode function, so I have to wrap and and create a decode func that can be accepted by RegisterNotificationsProtocol. Take a look at `network.MessageDecoder`